pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -168,3 +159,58 @@ func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidat
 
 	return deletedLS, async, err
 }
+
+// Update overrides the generic behavior of the Storage and writes first to the legacy storage and then to US.
+// Update overrides the behavior of the generic DualWriter and writes first to LegacyStorage and then to Storage.
+func (d *DualWriterMode2) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.Legacy.(rest.Updater)
+	if !ok {
+		return nil, false, errDualWriterUpdaterMissing
+	}
+
+	// get old and new (updated) object so they can be stored in legacy store
+	old, err := d.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.FromContext(ctx).Error(err, "could not get object", "mode", Mode2)
+			return nil, false, err
+		}
+		klog.FromContext(ctx).Error(err, "object not found for update, creating one", "mode", Mode2)
+		forceAllowCreate = true
```

Comment:
Depending on which of the approaches you decide to take from https://github.com/grafana/grafana/pull/85844/files#r1562844729, this may need to be removed? I'm not familiar with this field.

Actually I don't think there's any implementation in place to allow this field to take effect. I think we need to be making a Create call here instead of setting forceAllowCreate. So something similar to this: https://github.com/grafana/grafana/blob/main/pkg/services/apiserver/storage/entity/storage.go#L574-L591

If I'm mistaken and forceAllowCreate does work, then we would want to set it for the call to Storage but not the one to LegacyStorage. If an update call is being made to a nonexistent LegacyStorage object than that's an error.

Reply:
ok, I removed the flag. Apparently, `UpdatedObject` will already create an object if none is found, so I think we are good.

Reply:
there was actually a bug caught by the newly introduced tests if an object is created anew :)